tests/cypress.support.js

```diff
@@ -117,43 +117,82 @@ Cypress.Commands.add("continueOn", function(cluster, expression) {
     })
 })
 
-// Available as cy.dexStaticLogin() requires dex static login to be enabled
-Cypress.Commands.add("dexStaticLogin", function() {
-  // Check if enabled
+// Available as cy.dexStaticUserLogin() requires dex static login to be enabled
+Cypress.Commands.add("dexStaticUserLogin", function(username, loginUrl, cookieName) {
+  // Check if static login is enabled
   cy.yqDig("sc", ".dex.enableStaticLogin")
     .then(staticLoginEnabled => {
       if (staticLoginEnabled !== "true") {
         cy.fail("dexStaticLogin: requires dex static login to be enabled")
       }
     })
 
-  // Conditionally skip connector selection
-  cy.yqSecrets(".dex.connectors | length")
-    .then(connectors => {
-      if (connectors !== "0") {
-        cy.contains("Log in with Email")
-          .click()
+  cy.session([username, loginUrl, cookieName], () => {
+    cy.visit(loginUrl)
+
+    // this is needed for Grafana
+    cy.title().then(($title) => {
+      if ($title === 'Grafana') {
+        cy.contains("Sign in with dex").click()
       }
     })
 
-  // Fetch and type in credentials
-  // The selection of the input is a bit weak but the field's label is in its own div
-  cy.yqSecrets(".dex.staticPasswordNotHashed")
-    .then(password => {
-      cy.contains("Email Address")
-        .parent()
-        .parent()
-        .find("input")
-        .type("admin@example.com", { log: false })
-
-      cy.contains("Password")
-        .parent()
-        .parent()
-        .find("input")
-        .type(password, { log: false })
+    // Conditionally skip connector selection
+    cy.yqSecrets(".dex.connectors | length")
+      .then(connectors => {
+        if (connectors !== "0") {
+          cy.contains('button', 'Log in with Email').click()
+        }
+      })
+    cy.get('input#login').type(username)
+    cy.yqSecrets(".dex.staticPasswordNotHashed")
+      .then(password => {
+        // {enter} causes the form to submit
+        cy.get('input#password').type(`${password}{enter}`, { log: false })
+      })
+    // this is needed for Harbor first time login
+    cy.title().then(($title) => {
+      if ($title === 'Harbor') {
+        cy.wait(1000)
+        cy.get("body").then(($body) => {
+          const hasOidcBanner = $body[0].querySelectorAll("*[class='modal-title oidc-header-text']")
+          cy.log(hasOidcBanner)
+          if (hasOidcBanner.length > 0) {
+            cy.get('input[name=oidcUsername]').clear().type('cypress_static_user')
+            cy.contains('button', 'SAVE').click()
+          }
+        })
+      }
     })
+    // Ensure Auth0 has redirected us back to the original URL
+    cy.url().should('include', loginUrl)
+    cy.getCookie(cookieName).should('exist')
+  },
+  )
```

Comment:
Is this the end of the session call? Then it should be possible to join `})` on one line.